src/Dev/SapphireTest.php

```diff
@@ -1212,16 +1214,30 @@ public static function createInvalidArgumentException($argument, $type, $value =
     }
 
     /**
-     * Returns the annotations for this test.
-     *
-     * @return array
+     * Returns the annotations for this test
      */
     public function getAnnotations(): array
     {
-        return TestUtil::parseTestMethodAnnotations(
-            get_class($this),
-            $this->getName(false)
-        );
+        $class = get_class($this);
+        $method = $this->name();
+        $ret = [];
+        foreach (['method', 'class'] as $what) {
+            if ($what === 'method') {
+                $reflection = new ReflectionMethod($class, $method);
+            } else {
+                $reflection = new ReflectionClass($class);
+            }
+            preg_match_all('#@(.*?)\n#s', $reflection->getDocComment(), $annotations);
+            $ret[$what] = [];
+            foreach ($annotations[1] as $annotation) {
+                $parts = explode(' ', $annotation);
+                $key = array_shift($parts);
+                $value = implode(' ', $parts);
+                $ret[$what][$key] ??= [];
+                $ret[$what][$key][] = $value;
+            }
+        }
+        return $ret;
```

Comment:
I would have expected `nikic/php-parser` to be able to do this? Can we leverage that package to handle this for us instead?

Reply:
As far as a can tell php-parser will also just return the DocComment - https://github.com/nikic/PHP-Parser/blob/master/lib/PhpParser/Comment/Doc.php - it won't get any more finely grained than that and extract annotations for us.

Given that PHP natively has a getDocComment() method available, seems like there's no benefit to using php-parser for this scenario

Reply:
That's a shame but fair enough. Thanks for taking a look.